components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1931,8 +1930,8 @@ private AccessTokenExtendedAttributes getExtendedTokenAttributes(OAuthMessage oA
         try {
             ServiceProvider serviceProvider = getServiceProvider(oauth2Params.getClientId());
             // TODO: Improve to read the script separately instead of reading from adaptive script.
-            if (!EndpointUtil.isExternalConsentPageEnabledForSP(serviceProvider) ||
-                    serviceProvider.getLocalAndOutBoundAuthenticationConfig().getAuthenticationScriptConfig() == null) {
+            if (!serviceProvider.getLocalAndOutBoundAuthenticationConfig().getAuthenticationScriptConfig()
```

Comment:
NPE will occur with there is no authentication script config?

Reply:
Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2437/commits/007afb081cec57305d9c7387bb5599d49b9b148e